app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action ->{load_user params.dig(:password_reset, :email)},
+                only: :create
+  before_action ->{load_user params[:email]}, only: %i(edit update)
+  before_action :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params.dig(:password_reset, :email)&.downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "reset_password.sent_email"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if user_params[:password].blank?
+      @user.error.add :password, t("error")
+      render :edit
+    elsif @user.update user_params
+      log_in @user
+      @user.update_column :reset_digest, nil
+      flash[:success] = t "reset_password.success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def load_user email
+    @user = User.find_by email: email&.downcase
```

Comment:
```suggestion
    @user = User.find_by email: (params.dig(:password_reset, :email) || params.dig(:email))&.downcase
```
ri khỏe hơn nè =)) nếu chỉ dùng trong mỗi controller này